### ISSUE TITLE
Ctrl+C does not work in the shell

### ISSUE THREAD

**Issue by czardoz** [czardoz/hornet#10]
This functionality is essential for commands like ping, where a user interrupt is necessary to quit.

**Comment by czardoz** [comment]
The ``inputcooker_store_queue(self, char)`` method can be overriden to detect when Ctrl+C is pressed:

```
    def inputcooker_store_queue(self, char):
        """Put the cooked data in the input queue (no locking needed)"""
        if type(char) in [type(()), type([]), type("")]:
            for v in char:
                if v == chr(3):
                    self.interrupt_callback()
                self.cookedq.put(v)
        else:
            self.cookedq.put(char)
```

Here, ``self.interrupt_callback`` should be a function that stop the execution of the current command.

**Comment by czardoz** [comment]
Implemented a different approach, where a flag, ``shell.interrupt`` is set to ``True`` whenever Ctrl+C is read. 

This was fixed in 2c16dbc8e72a89303ff58a1b01cd56977b15dcb0
